File: scala-3.6/template-renderer/src/main/scala/com/rbleggi/templaterenderer/renderers/HTMLRenderer.scala
package com.rbleggi.templaterenderer.renderers

import com.rbleggi.templaterenderer.core.TemplateRenderer

class HTMLRenderer extends TemplateRenderer {
  override def render(data: Map[String, String]): Array[Byte] =
    println("Generating HTML Template...")
    val htmlContent =
      s"""
         |<html>
         |<head><title>${data("title")}</title></head>
         |<body>
         |<h1>${data("title")}</h1>
         |<p>${data("content")}</p>
         |</body>
         |</html>
         |""".stripMargin
    println("HTML Template Generated Successfully!")
    htmlContent.getBytes()
}

File: kotlin/template-renderer/src/main/kotlin/com/rbleggi/templaterenderer/service/renderer/HtmlRenderer.kt
package com.rbleggi.templaterenderer.service.renderer

class HtmlRenderer : TemplateRenderer() {
    override fun renderTitle(title: String): String = "<h1>$title</h1>"

    override fun renderContent(content: String): String = "<p>$content</p>"

    override fun assemble(title: String, content: String): ByteArray =
        "<html><body>\n$title\n$content\n</body></html>".toByteArray()
}
